vscode/src/chat/chat-view/ChatsController.ts

```diff
@@ -0,0 +1,479 @@
+        }
+
+        this.supportTreeViewProvider.syncAuthStatus(authStatus)
+
+        this.panel.syncAuthStatus()
+    }
+
+    public async restoreToPanel(panel: vscode.WebviewPanel, chatID: string): Promise<void> {
+        try {
+            await this.getOrCreateEditorChatController(chatID, panel.title, panel)
+        } catch (error) {
+            logDebug('ChatsController', 'restoreToPanel', { error })
+
+            // When failed, create a new panel with restored session and dispose the old panel
+            await this.getOrCreateEditorChatController(chatID, panel.title)
+            panel.dispose()
+        }
+    }
+
+    public registerViewsAndCommands() {
+        this.disposables.push(
+            vscode.window.registerTreeDataProvider(
+                'cody.support.tree.view',
+                this.supportTreeViewProvider
+            ),
+            vscode.window.registerWebviewViewProvider('cody.chat', this.panel, {
+                webviewOptions: { retainContextWhenHidden: true },
+            })
+        )
+
+        const debouncedRestoreToEditor = debounce(
+            async (chatID: string, chatQuestion?: string) => {
+                try {
+                    logDebug('ChatsController', 'debouncedRestorePanel')
+                    await this.getOrCreateEditorChatController(chatID, chatQuestion)
+                } catch (error) {
+                    logDebug('ChatsController', 'debouncedRestorePanel', 'failed', error)
+                }
+            },
+            250,
+            { leading: true, trailing: true }
+        )
+        const debouncedCreateNewWebviewPanel = debounce(
+            () => this.getOrCreateEditorChatController(),
+            250,
+            {
+                leading: true,
+                trailing: true,
+            }
+        )
+
+        this.disposables.push(
+            vscode.commands.registerCommand(
+                'cody.chat.moveToEditor',
+                async () => await this.moveChatFromPanelToEditor()
+            ),
+            vscode.commands.registerCommand(
+                'cody.chat.moveFromEditor',
+                async () => await this.moveChatFromEditorToPanel()
+            ),
+            vscode.commands.registerCommand('cody.action.chat', args =>
+                this.submitChatInNewEditor(args)
+            ),
+            vscode.commands.registerCommand('cody.chat.focusView', () =>
+                vscode.commands.executeCommand('cody.chat.focus')
+            ),
+            vscode.commands.registerCommand('cody.chat.signIn', () =>
+                vscode.commands.executeCommand('cody.chat.focus')
+            ),
+            vscode.commands.registerCommand('cody.chat.newPanel', async () => {
+                await this.panel.clearAndRestartSession()
+                await vscode.commands.executeCommand('cody.chat.focus')
+            }),
+            vscode.commands.registerCommand('cody.chat.newEditorPanel', debouncedCreateNewWebviewPanel),
```

Comment:
I saw the agent tests are failing because of this command but haven't looked into the cause yet. Making the following change fixed the error for me though so something about the debounce logic?

```suggestion
            vscode.commands.registerCommand('cody.chat.newEditorPanel', () =>
                this.getOrCreateEditorChatController()
            ),
```